pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/ResourceLockImpl.java

```diff
@@ -71,7 +72,20 @@ public synchronized T getValue() {
 
     @Override
     public synchronized CompletableFuture<Void> updateValue(T newValue) {
-       return acquire(newValue);
+        // If there is an operation in progress, we're going to let it complete before attempting to
+        // update the value
+        pendingOperationFuture = pendingOperationFuture.thenCompose(v -> {
```

Comment:
`pendingOperationFuture` may throw exceptions `LockBusyException`, `BadVersionException` or other connectivity issues and return errors directly. Do we need to check that the relevant usage logic has handled this exception?

Reply:
That's correct, I think we're missing to reset it once it gets completed once.